### PR TITLE
Docs: Changes corresponding to docs design updates

### PR DESCRIPTION
**Pull request by kylegach**
## What I did

See also: https://github.com/storybookjs/frontpage/pull/626

- Remove emoji from and simplify top-level titles
- Remove concept of `bullet-link`
- Remove `/<renderer>` from URLs in the CLI templates
- Remove `/<renderer>` from URLs in the docs
- Update TOC to use `heading` type
    - Rename `introduction.md`, `overview.md`, `how-to-contribute.md` pages -> `index.md` pages
    - Add all-new `index.md` pages for Sharing and API
    - Add `redirectPath` to `get-started` TOC item
        - Temporary workaround until we build a proper docs homepage
    - Find/replace `introduction.md`/`overview.md` -> `index.md`
    - Find/replace `/introduction`/`/overview` -> `/`
    - Add `hideRendererSelector: true` to frontmatter of (some) pages that aren't conditional on renderer

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/frameworks/ember/template/cli/Button.stories.js
Comment (jonniebigodes) [inline code/frameworks/ember/template/cli/Button.stories.js:5]: @kylegach quick question, do we need to preserve the Ember framework here? And if so, shouldn't the link below use it as well for consistency?

File: code/addons/actions/README.md
Comment (jonniebigodes) [inline code/addons/actions/README.md:11]: @kylegach, one small item here. I suggest removing the framework in the links for our addons and providing with the installation process. Reasoning here: if the reader checks this but is working with Vue or has checked our Vue docs, they should not see a framework that's not his own. This leads him to have to go up the page and select another renderer just to see what wants to read about, otherwise it should automatically default to React. 
Just trying to keep it consistent, and I'll leave it to you to proceed.

**Review by jonniebigodes — APPROVED** [review]
Thanks, @kylegach, for putting this pull request together and significantly improving the documentation. I've checked a few small items for you to consider when you have a moment. To unblock you, I'm approving the pull request.

File: code/frameworks/ember/template/cli/Button.stories.js
Comment (kylegach) [reply code/frameworks/ember/template/cli/Button.stories.js:5]: Good catch! No, we do not. I simply didn't include `ember` links in my Find/Replace'ing.

File: code/addons/actions/README.md
Comment (kylegach) [reply code/addons/actions/README.md:11]: This PR corresponds with one in frontpage that completely removes the renderer from the URL. URLs containing a renderer will redirect to the the correct renderer-less URL with zero affect on whatever renderer is in their localStorage and used for the site.

Comment (kylegach) [reply code/addons/actions/README.md:11]: Additionally, I opted to _not_ update the various URLs in READMEs in this PR. While most renderers only have a few results, `react` has 150+ and I didn't want to make this one any bigger than it needed to be.

**Pull request merged**
